### PR TITLE
samples/drivers/mbox: Convert to sysbuild, add support for nrf5340bsim and add runtime test

### PR DESCRIPTION
**Pull request by aescolar**
Convert this sample application build to sysbuild.
Add support for this sample on the simulated nrf5340
Add a runtest for the simulated target.

### CODE REVIEW

**Comment by kartben** [comment]
@carlocaione should there be an mbox maintainer area?

File: samples/drivers/mbox/Kconfig.sysbuild [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+source "share/sysbuild/Kconfig"
+
+config REMOTE_BOARD
+string
+	default "nrf5340dk_nrf5340_cpunet" if $(BOARD) = "nrf5340dk_nrf5340_cpuapp"
+	default "nrf5340bsim_nrf5340_cpunet" if $(BOARD) = "nrf5340bsim_nrf5340_cpuapp"
+	default "adp_xc7k_ae350" if $(BOARD) = "adp_xc7k_ae350"
+	default "nrf5340dk_nrf5340_cpunet" if $(BOARD) = "mimxrt595_evk_cm33"
```
Comment (nordicjm) [inline samples/drivers/mbox/Kconfig.sysbuild:12]: this doesn't look right, how can the nrf5340 be the remote core of an nxp board?

Comment (aescolar) [reply samples/drivers/mbox/Kconfig.sysbuild:12]: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/mbox/CMakeLists.txt#L15-L16

Comment (nordicjm) [reply samples/drivers/mbox/Kconfig.sysbuild:12]: @yclistanford did you make a mistake in your PR?

**Review by nordicjm — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**